augur/validate.py

```diff
@@ -70,6 +70,10 @@ def validate(jsonToValidate, schema, filename):
             print("\tERROR: {}. Trace: {}".format(error.message, " - ".join(trace)), file=sys.stderr)
         raise ValidateError("Validation of {!r} failed.".format(filename))
 
+
+validate = validate_json  # TODO update uses and drop this alias
```

Comment:
Yes, for readability in the new modules, in `validate.py`, and anywhere else it may be used in the future. `augur.validate.validate_json` is more self-documenting, whereas `augur.validate.validate` doesn't give as many clues about programmer intent.

Relatedly, `from augur.validate import validate` prevents use of the name `validate` in any other module that needs to do more than one kind of validation.

---

The name `validate_json` still has room for improvement—it sounds like it just checks for valid JSON, which is not the full story. The method actually validates some JSON against a given schema. We could keep chasing an ideal name (`validate_json_against_schema`?), but at some point the verbosity becomes worse than the ambiguity of the short name.

My preferred solution would be to introduce a new class `JsonSchema`, which would subsume `validator.load_json_schema` and `validator.validate_json` (making `..._against_schema` implied by the class name). Then `validator.py` could just do something like this (very pseudocode):

```python
def auspice_config_v2(config_json, **kwargs):
    schema = JsonSchema("schema-auspice-config-v2.json")
    return schema.validate_json(config_json)

def export_v2(main_json, **kwargs):
    schema = JsonSchema("schema-export-v2.json")

    # if main_json.endswith("frequencies.json") ...
    #     ...

    return schema.validate_json(main_json)


def export_v1(meta_json, tree_json, **kwargs):
    meta_schema = JsonSchema("schema-export-v1-meta.json")
    tree_schema = JsonSchema("schema-export-v1-tree.json")

    # ...

    return meta_schema.validate_json(meta_json) and tree_schema.validate_json(tree_json)
```

But that's a change for another PR. `validate` → `validate_json` is a minor change that's easy to include for a small amount of immediate value.

Reply:
I like this idea a lot, although I agree it's definitely for another PR. :)